tests/external_botocore/test_boto3_kinesis.py

```diff
@@ -0,0 +1,237 @@
+    validate_transaction_metrics,
+)
+
+from newrelic.api.background_task import background_task
+from newrelic.common.package_version_utils import get_package_version_tuple
+from newrelic.hooks.external_botocore import CUSTOM_TRACE_POINTS
+
+MOTO_VERSION = get_package_version_tuple("moto")
+BOTOCORE_VERSION = get_package_version_tuple("boto3")
+
+URL = "kinesis.us-east-1.amazonaws.com"
+TEST_STREAM = f"python-agent-test-{uuid.uuid4()}"
+EXPECTED_AGENT_ATTRS = {
+    "exact_agents": {
+        "cloud.platform": "aws_kinesis_data_streams",
+        "cloud.resource_id": f"arn:aws:kinesis:us-east-1:123456789012:stream/{TEST_STREAM}",
+    },
+}
+
+AWS_ACCESS_KEY_ID = "AAAAAAAAAAAACCESSKEY"
+AWS_SECRET_ACCESS_KEY = "AAAAAASECRETKEY"  # nosec
+AWS_REGION = "us-east-1"
+
+_kinesis_scoped_metrics = [
+    (f"MessageBroker/Kinesis/Stream/Produce/Named/{TEST_STREAM}", 2),
+    (f"MessageBroker/Kinesis/Stream/Consume/Named/{TEST_STREAM}", 1),
+    (f"Kinesis/create_stream/{TEST_STREAM}", 1),
+    (f"Kinesis/describe_stream/{TEST_STREAM}", 1),
+    (f"Kinesis/get_shard_iterator/{TEST_STREAM}", 1),
+    (f"Kinesis/delete_stream/{TEST_STREAM}", 1),
+    (f"External/{URL}/botocore/POST", 2),
+]
+if BOTOCORE_VERSION < (1, 29, 0):
+    _kinesis_scoped_metrics = [
+        (f"MessageBroker/Kinesis/Stream/Produce/Named/{TEST_STREAM}", 2),
+        (f"Kinesis/create_stream/{TEST_STREAM}", 1),
+        (f"Kinesis/describe_stream/{TEST_STREAM}", 1),
+        (f"Kinesis/get_shard_iterator/{TEST_STREAM}", 1),
+        (f"Kinesis/delete_stream/{TEST_STREAM}", 1),
+        (f"External/{URL}/botocore/POST", 4),
+    ]
+
+_kinesis_rollup_metrics = [
+    (f"MessageBroker/Kinesis/Stream/Produce/Named/{TEST_STREAM}", 2),
+    (f"MessageBroker/Kinesis/Stream/Consume/Named/{TEST_STREAM}", 1),
+    (f"Kinesis/create_stream/{TEST_STREAM}", 1),
+    (f"Kinesis/describe_stream/{TEST_STREAM}", 1),
+    (f"Kinesis/get_shard_iterator/{TEST_STREAM}", 1),
+    (f"Kinesis/delete_stream/{TEST_STREAM}", 1),
+    ("External/all", 4),
+    ("External/allOther", 4),
+    (f"External/{URL}/all", 2),
+    (f"External/{URL}/botocore/POST", 2),
+]
+if BOTOCORE_VERSION < (1, 29, 0):
+    _kinesis_rollup_metrics = [
+        (f"MessageBroker/Kinesis/Stream/Produce/Named/{TEST_STREAM}", 2),
+        (f"Kinesis/create_stream/{TEST_STREAM}", 1),
+        (f"Kinesis/describe_stream/{TEST_STREAM}", 1),
+        (f"Kinesis/get_shard_iterator/{TEST_STREAM}", 1),
+        (f"Kinesis/delete_stream/{TEST_STREAM}", 1),
+        ("External/all", 4),
+        ("External/allOther", 4),
+        (f"External/{URL}/all", 4),
+        (f"External/{URL}/botocore/POST", 4),
+    ]
+
+_kinesis_scoped_metrics_error = [
+    ("MessageBroker/Kinesis/Stream/Produce/Named/Unknown", 1),
+]
+
+_kinesis_rollup_metrics_error = [
+    ("MessageBroker/Kinesis/Stream/Produce/Named/Unknown", 1),
+]
+
+
+@background_task()
+@mock_aws
+def test_instrumented_kinesis_methods():
```

Comment:
Not proud of this test but not sure how to implement it otherwise. Open to ideas.